app/controllers/admin/base_controller.rb

```diff
@@ -0,0 +1,7 @@
+class Admin::BaseController < ApplicationController
+  before_action :require_admin
+
+  def require_admin
+    render file: '/public/404' unless current_admin?
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.